### DISCUSSION TITLE
Latest image 'mcr.microsoft.com/dotnet/aspnet:8.0' still has the HIGH vulnerability CVE-2023-50387

### DISCUSSION THREAD

**Discussion by klingu** [dotnet/dotnet-docker#5357] [Q&A] [answered]
Hi...,

I build an application based on the image 'mcr.microsoft.com/dotnet/aspnet:8.0'. When I shipped it to the company who hosts this container, they scanned the image with a 'High' vulnerability.

In the image of Microsoft 'mcr.microsoft.com/dotnet/aspnet:8.0' is a high vulnerability. The image is based on a Debian Bookworm distribution. The [vulnerability](https://security-tracker.debian.org/tracker/CVE-2023-50387) is in the 'systemd', version: 252.22-1~deb12u1 There is a [fix](https://packages.debian.org/sid/amd64/systemd-resolved/download) of this package in version: 255.4-1

Is there a fix coming soon? Or is there a way how I can fix this myself?
If so, how do I start the 'mcr.microsoft.com/dotnet/aspnet:8.0' image and install the newer systemd package.

What I've tried and did not help or solved the vulnerability:

- in the docker file: RUN apt-get update -yy && apt-get upgrade -yy
- in the docker file: apt-get install systemd=255.4-1+b1 (because this version is not in the package list)
- wget is not available in the docker file and not in the running container.
- curl is not available in the docker file and not in the running container.

Thanks in advance for you help.

Jeffrey

**Comment by mthalman** [comment, marked as answer]
.NET relies on the Linux distro maintainers to update their packages. The fixed package version is not yet available for Bookworm, as indicated in [CVE-2023-50387](https://security-tracker.debian.org/tracker/CVE-2023-50387):

![image](https://github.com/dotnet/dotnet-docker/assets/15789599/c62fa449-1f61-4c95-bf85-ad886490e3a0)

We can make no recommendation of other solutions. As I mentioned, we rely on the distro maintainers here as they have a process to ensure correctness and compatibility when integrating new package versions into each distro version.

You may want to explore other distros we provide like Alpine or our [Ubuntu Chiseled](https://github.com/dotnet/dotnet-docker/blob/main/documentation/ubuntu-chiseled.md) images which have a lower attack surface due to their smaller set of packages installed.

**Reply by klingu** [reply]
With the Alpine image (mcr.microsoft.com/dotnet/aspnet:8.0-alpine3.18-amd64 AS base) I run into an other issue: I can not create a Non-Root-user like:

```
ARG USERNAME=me-myself-and-I
ARG USER_UID=xxxx
ARG USER_GID=xxxx

RUN groupadd --gid $USER_GID $USERNAME \
    && useradd --uid $USER_UID --gid $USER_GID -M $USERNAME
```

So I ended up with the Ubuntu image: **mcr.microsoft.com/dotnet/aspnet:8.0-jammy-amd64 AS base**

This solved my issue.

Thanks for the pointing out!

**Reply by richlander** [reply]
You know that all .NET 8 images contain a non-root user?

https://devblogs.microsoft.com/dotnet/securing-containers-with-rootless/

FYI: The UID has changed (to `1654`) since that post. I need to update the post.

**Comment by richlander** [comment]
You can install `curl` or `wget` if you need those tools. However, the package you referenced is from `sid`. Installing it on `bookworm` is not supported and has undefined behavior.

**Reply by klingu** [reply]
The Debian [download site](https://packages.debian.org/sid/amd64/systemd-resolved/download) says: Download Page for systemd-resolved_255.4-1+b1_amd64.deb on AMD64 machines

But this is probably a kind of misleading ;-)

Thanks for your quick reply!

**Reply by richlander** [reply]
Respectfully, It's not misleading. You are on a `sid` page, which is indicated in multiple places (in the URL and the page).
